Terminal.Gui/Core/View.cs

```diff
@@ -446,70 +446,102 @@ void SetTabIndex ()
 		/// </para>
 		/// </remarks>
 		public virtual Rect Frame {
 			get => frame;
 			set {
 				frame = new Rect (value.X, value.Y, Math.Max (value.Width, 0), Math.Max (value.Height, 0));
+				X = frame.X;
+				Y = frame.Y;
+				Width = frame.Width;
+				Height = frame.Height;
 				TextFormatter.Size = GetSizeNeededForTextAndHotKey ();
 				SetNeedsLayout ();
 				SetNeedsDisplay ();
 			}
 		}
 
-		///// <summary>
-		///// Gets an enumerator that enumerates the subviews in this view.
-		///// </summary>
-		///// <returns>The enumerator.</returns>
-		//public IEnumerator GetEnumerator ()
-		//{
-		//	foreach (var v in InternalSubviews)
-		//		yield return v;
-		//}
-
 		LayoutStyle layoutStyle;
 
 		/// <summary>
-		/// Controls how the View's <see cref="Frame"/> is computed during the LayoutSubviews method, if the style is set to
-		/// <see cref="Terminal.Gui.LayoutStyle.Absolute"/>, 
-		/// LayoutSubviews does not change the <see cref="Frame"/>. If the style is <see cref="Terminal.Gui.LayoutStyle.Computed"/>
-		/// the <see cref="Frame"/> is updated using
-		/// the <see cref="X"/>, <see cref="Y"/>, <see cref="Width"/>, and <see cref="Height"/> properties.
+		/// Gets or sets what <see cref="Terminal.Gui.LayoutStyle"/> the view is using. 
 		/// </summary>
+		/// <remarks>
+		/// <para>
+		/// Setting this property to <see cref="Terminal.Gui.LayoutStyle.Absolute"/> will cause <see cref="Frame"/> to determine the size and position of the view. <see cref="X"/> and <see cref="Y"/> will be 
+		/// set to <see cref="Dim.DimAbsolute"/> using <see cref="Frame"/>.
+		/// </para>
+		/// <para>
+		/// Setting this property to <see cref="Terminal.Gui.LayoutStyle.Computed"/> will cause the view to use the <see cref="LayoutSubviews"/> method to 
+		/// size and position of the view. If either of the <see cref="X"/> and <see cref="Y"/> properties are `null` they will be set to <see cref="Pos.PosAbsolute"/> using
+		/// the current value of <see cref="Frame"/>. 
+		/// If either of the <see cref="Width"/> and <see cref="Height"/> properties are `null` they will be set to <see cref="Dim.DimAbsolute"/> using <see cref="Frame"/>.
+		/// </para>
+		/// </remarks>
 		/// <value>The layout style.</value>
 		public LayoutStyle LayoutStyle {
-			get => layoutStyle;
+			get {
+				if ((X == null || X is Pos.PosAbsolute) &&  (Y == null || Y is Pos.PosAbsolute) && 
+					(Width == null || Width is Dim.DimAbsolute) && (Height == null || Height is Dim.DimAbsolute)) {
+					return LayoutStyle.Absolute;
+				}
+				return LayoutStyle.Computed;
+			}
 			set {
-				layoutStyle = value;
+				switch (value) {
+				case LayoutStyle.Absolute:
+					X = Frame.X;
+					Y = Frame.Y;
+					Width = Frame.Width;
+					Height = Frame.Height;
+					break;
+					
+				case LayoutStyle.Computed:
+					if (X == null) X = Frame.X;
+					if (Y == null) Y = Frame.Y;
+					if (Width == null) Width = Frame.Width;
+					if (Height == null) Height = Frame.Height;
```

Comment:
Can be replaced by:
```cs
X ??= Frame.X;
Y ??= Frame.Y;
Width ??= Frame.Width;
Height ??= Frame.Height;
```
Additionally in each` Pos/Dim (X, Y, Width, Height)` setter you after must update the `_layoutStyle` field by get the `LayoutStyle` value:
`
_layoutStyle = LayoutStyle;
`